src/main/java/at/hannibal2/skyhanni/features/rift/area/mountaintop/SunGeckoHelper.kt

```diff
@@ -0,0 +1,291 @@
+import at.hannibal2.skyhanni.events.MobEvent
+import at.hannibal2.skyhanni.events.ScoreboardUpdateEvent
+import at.hannibal2.skyhanni.events.skyblock.ScoreboardAreaChangeEvent
+import at.hannibal2.skyhanni.features.rift.RiftAPI
+import at.hannibal2.skyhanni.mixins.hooks.RenderLivingEntityHelper
+import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
+import at.hannibal2.skyhanni.utils.ColorUtils.addAlpha
+import at.hannibal2.skyhanni.utils.RegexUtils.findMatcher
+import at.hannibal2.skyhanni.utils.RegexUtils.groupOrNull
+import at.hannibal2.skyhanni.utils.RenderUtils.renderStrings
+import at.hannibal2.skyhanni.utils.SimpleTimeMark
+import at.hannibal2.skyhanni.utils.StringUtils.allLettersFirstUppercase
+import at.hannibal2.skyhanni.utils.TimeUtils
+import at.hannibal2.skyhanni.utils.TimeUtils.format
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraft.init.Blocks
+import net.minecraft.item.Item
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import java.awt.Color
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.milliseconds
+import kotlin.time.Duration.Companion.seconds
+
+@SkyHanniModule
+object SunGeckoHelper {
+    val config: SunGeckoConfig get() = SkyHanniMod.feature.rift.area.mountaintop.sunGecko
+    val pos: Position get() = config.pos
+    var display: ArrayList<String> = ArrayList()
+    var modifiers: MutableSet<MODIFIERS> = mutableSetOf()
+    private val patternGroup = RepoPattern.group("rift.area.mountaintop.sungecko")
+
+    /**
+     * REGEX-TEST: §a[⬛⬛⬛⬜§c⬜ §e§lx2 §c⬜⬜⬜⬜⬜]
+     * REGEX-TEST: §c[⬜⬜⬜⬜⬜ §e§lx1 §c⬜⬜⬜⬜⬜]
+     * REGEX-TEST: §a[⬛⬛⬛⬛§c⬛ §e§lx1 §c⬛⬜⬜⬜⬜]
+     * REGEX-TEST: §a[⬛⬛⬛⬛⬛ §e§lx1 §a⬛§c⬛⬛⬜⬜]
+     * REGEX-TEST: §a[⬛⬛⬛⬛⬛ §e§lx1 §c§c⬛⬛⬜⬜⬜]
+     * REGEX-TEST: §a[⬛⬛⬛§c⬛⬛ §e§lx2 §c⬛⬛⬛⬜⬜]
+     */
+    private val sunGeckoActionBar by patternGroup.pattern(
+        "actionbar",
+        "(?<firstHalf>§[ac]\\[.*) §e§lx(?<combo>\\d+) (?<secondHalf>§[ac].*)]",
+    )
+
+    private var healthLeft = 250
+    private var totalHealth = 250
+    private var actionBarFormatted: String = ""
+    private var onFirstPhase = true
+    private var timeSliceDuration = Duration.ZERO
+    private var timeSinceLastHit = SimpleTimeMark.farPast()
+    private var combo = 1
+    private var scanningChat = false
+    private var inTimeChamber = false
+    private var currentBoss: Mob? = null
+
+    fun reset() {
+        display.clear()
+        healthLeft = 250
+        totalHealth = 250
+        actionBarFormatted = ""
+        onFirstPhase = true
+        timeSliceDuration = Duration.ZERO
+        timeSinceLastHit = SimpleTimeMark.farPast()
+        combo = 1
+        inTimeChamber = false
+        currentBoss = null
+    }
+
+    fun updateDisplay() {
+        display.clear()
+        var displayHealthLeft = healthLeft
+        var displayTotalHealth = totalHealth
+        if (modifiers.contains(MODIFIERS.REVIVAL)) {
```

Comment:
we usually use the contains operator instead of the actual contains function, like
```suggestion
        if (MODIFIERS.REVIVAL in modifiers) {
```